### DISCUSSION TITLE
how to do a query/passage ranking task in question answering using t5

### DISCUSSION THREAD

**Discussion by vkaul11** [google-research/text-to-text-transfer-transformer#787] [Q&A]
I have a task in which we have a query : "What is the best way to lose weight" and I have different candidate passages. Based on coarse retrieval score we create positive (where there is some relevance above a threshold) and negative examples (very low relevance) and we train a BERT like model with sentence A/ sentence B format Next Sentence Prediction task. As the model can emit a score between 0 and 1, I can always use the score for ranking.
However in t5 if I have the Input like this
`Input: Question"What is the best way to lose weight" Response "Using Keto Diet"   `  `Output: 1`
`Input: Question"What is the best way to lose weight" Response "Barack Obama"   `  `Output: 0`
How exactly do we ensure that the outputs are scores between 0 and 1? Is there some other way to handle this problem through t5?

**Comment by craffel** [comment]
The best approach is to "score" the different options and use that for ranking. There is past work on this, see e.g. https://arxiv.org/abs/2003.06713
